File: src/features/typing/data/ts/group_anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
  const groups = new Map<string, string[]>();
  for (const s of strs) {
    const key = [...s].sort().join('');
    const group = groups.get(key);
    if (group === undefined) {
      groups.set(key, [s]);
    } else {
      group.push(s);
    }
  }
  return [...groups.values()];
}

File: src/features/typing/data/dart/group_anagrams.dart
class Solution {
  List<List<String>> groupAnagrams(List<String> strs) {
    final groups = <String, List<String>>{};
    for (final s in strs) {
      final key = (s.split('')..sort()).join();
      groups.putIfAbsent(key, () => []).add(s);
    }
    return groups.values.toList();
  }
}
